.github/workflows/test.yaml

```diff
@@ -111,36 +129,59 @@ jobs:
   docker-lint:
     name: Dockerfile lint
     runs-on: ubuntu-22.04
+    outputs:
+      outcome: ${{ steps.report.outputs.outcome }}
+    defaults:
+      run:
+        working-directory: ${{ inputs.working-directory }}
     steps:
       - uses: actions/checkout@v3
       - name: Check for Dockerfiles
-        run: echo Dockerfiles=$(find . -name "*Dockerfile") >> $GITHUB_ENV
-      - if: ${{ env.Dockerfiles != '' }}
+        id: dockerfiles
+        run: echo "found=$(find . -name "*Dockerfile")" >> $GITHUB_OUTPUT
+      - if: ${{ steps.dockerfiles.outputs.found != '' }}
         name: Run HadoLint
         uses: jbergstroem/hadolint-gh-action@v1
         with:
-          dockerfile: "${{ env.Dockerfiles }}"
-  lint-metadata:
+          dockerfile: "${{ inputs.working-directory }}${{ steps.dockerfiles.outputs.found }}"
+      - name: Report
+        id: report
+        run: echo "outcome=$([ '${{ steps.dockerfiles.outputs.found }}' = '' ] && echo skip || echo success)" >> $GITHUB_OUTPUT
+  metadata-lint:
     name: Lint metadata.yaml
     runs-on: ubuntu-22.04
+    defaults:
+      run:
+        working-directory: ${{ inputs.working-directory }}
+    outputs:
+      outcome: ${{ steps.report.outputs.outcome }}
     steps:
       - uses: actions/checkout@v3
-      - name: Install check-jsonschema
-        run: python3 -m pip install check-jsonschema
-      - name: Download schema
-        run: curl -fLsSo $PWD/metadata.schema https://raw.githubusercontent.com/canonical/operator-workflows/main/.github/files/metadata.schema
+      - name: Check file existence
+        id: metadataYaml
+        run: echo "exists=$([ -f metadata.yaml ] && echo "true" || echo "false")" >> $GITHUB_OUTPUT
       - name: Run lint
-        run: check-jsonschema metadata.yaml --schemafile metadata.schema
+        if: steps.metadataYaml.outputs.exists == 'true'
+        run: |
+          python3 -m pip install check-jsonschema
+          curl -fLsSo $PWD/metadata.schema https://raw.githubusercontent.com/canonical/operator-workflows/main/.github/files/metadata.schema
```

Comment:
Didn't review the PR of Arturo who added this, but what if your project imports the workflow from another branch than main to test a new schema
Isn't there a way to know which branch was used to import the workflow and use this info ?

Reply:
That's not trivial for the reusable workflows. I didn't find a way to get the branch from this repository when called from another one. If you have any suggestion, it's welcome :)

Reply:
I think it would be possible although I propose to do it in a new PR, just checkout the operator-workflow repo to /tmp and copy/ link the schema file, the branch can be an optional parameter to the workflow. I propose we do that when we need to

Reply:
The difficulty is in finding the branch for this repository. The env vars will be populated with the source repository specifics